modules/programs/mpv.nix

```
@@ -50,7 +50,7 @@ let
     renderOptions { profile = concatStringsSep "," profiles; };
 
   mpvPackage = if cfg.scripts == [ ] then
-    pkgs.mpv
+    cfg.package
   else
     pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
```

Why not something like
```suggestion
    pkgs.wrapMpv pkgs.mpv-unwrapped (
      { scripts = cfg.scripts; }
      // cfg.extraWrapperArgs
    );
```
?

Seems to me that this would maintain backwards compatibility as well as allowing documentation of arguments chosen to be explicit options, like `scripts`.

For `mpv-unwrapped` overrides I think the `wrapMpv` should add an argument `mpvOverrides` or something so that

```nix
pkgs.wrapMpv pkgs.mpv-unwrapped {
  scripts = [ whatever ];
  mpvOverrides.vapoursynthSupport = true;
};
```

works as expected.

---

I would like to avoid this because this is not the first time this wrap changed and it may change in the future again. So I think it is better to expose people that needs more advanced needs to the package option itself.

This is why I initially preferred to simply removing the `scripts` option, but I think scripts is a more common case so maybe its makes sense to make it more accessible.

---

Fair enough. Perhaps something like
```
nixpkgs.overlays = [(self: super: {
  mpv-unwrapped = super.mpv-unwrapped.override {
    vapoursynthSupport = true;
  };
})];

programs.mpv = {
  …
  extraWrapperArgs.extraMakeWrapperArgs = [
    "--prefix" "LD_LIBRARY_PATH" ":" "${pkgs.vapoursynth-mvtools}/lib/vapoursynth"
  ];
];
```
would work?

---

Yeah, you could even overlay `mpv` itself and I think this works. But I think the advantaged of the `package` options is to make it easier to change, specially when the derivation is meant to be changed like `mpv`.

---

Hmm, fair point that `mpv` is meant to be changed. But I think that is more of an argument to incorporate that into the module, i.e., having explicit `enable` options for the most used arguments and a `mpvOverrides` escape hatch for the rest. That is, something like
```
programs.mpv = {
  …
  vapoursynthSupport = true;
  extraWrapperArgs.extraMakeWrapperArgs = [
    "--prefix" "LD_LIBRARY_PATH" ":" "${pkgs.vapoursynth-mvtools}/lib/vapoursynth"
  ];
}
```
in the vapoursynth case. Looking at the mpv package, the `vapoursynthSupport` option has been around for a bunch of years so it seems stable enough. It also has the benefit of allowing us to document these options a bit and making them somewhat discoverable compared to forcing people to look in the Nixpkgs source code.

I'm mainly wanting to avoid having something like
```
programs.mpv.package = (pkgs.wrapMpv (pkgs.mpv-unwrapped.override {
  vapoursynthSupport = true;
}) {
  extraMakeWrapperArgs = [
    "--prefix" "LD_LIBRARY_PATH" ":" "${pkgs.vapoursynth-mvtools}/lib/vapoursynth"
  ];
});
```
in the manual because it looks totally scary.

---

Ah, sorry! I just noticed that this got merged. I guess @berbiche disagrees with me, so I'll drop my arguing 🙂

---

If you want I can remove the `extraMakeWrapperArgs`, this is mostly my own use case that is complicated. An example like this in manual should be enough for most people:

```
programs.mpv.package = (pkgs.wrapMpv (pkgs.mpv-unwrapped.override {
  vapoursynthSupport = true;
}) {
  youtubeSupport = false;
});
```